lib/resources/wmi.rb

```diff
@@ -24,13 +25,20 @@ class WMI < Inspec.resource(1)
     include ObjectTraverser
     attr_accessor :content
 
-    def initialize(wmiclass, opts = {})
+    def initialize(wmiclass = nil, opts = nil)
+      # if we got a string, we handle deprecated behavior, class is opts[:class] now
+      if wmiclass.is_a? String
+        opts ||= {}
+        opts[:class] = wmiclass
+      else
+        # if we have one parameter, so wmiclass are the opts
+        opts ||= wmiclass
+        opts ||= {}
+      end
+
       # verify that this resource is only supported on Windows
       return skip_resource 'The `windows_feature` resource is not supported on your OS.' unless inspec.os.windows?
-
-      @wmiclass = wmiclass
-      @wminamespace = opts[:namespace]
-      @wmifilter = opts[:filter]
+      @options = opts
```

Comment:
Could this be shortened?
Maybe
```
@options = opts || {}
if wmiclass.is_a?(Hash)
  @options.merge!(wmiclass)
else
  @options[:class] = wmiclass
end
```
Should we add a deprecation warning for the old wmiclass mode?